Add CONTRIBUTING.MD file for contributing to docs (#11817)
github/codeql: docs/codeql/CONTRIBUTING.MD

# Contributing to CodeQL docs

We welcome contributions to our CodeQL docs. Want to improve existing docs or add new information you think would be helpful? Then please go ahead and open a pull request!

> ## Contributing to CodeQL docs on `docs.github.com`

Reviewer: Suggest making the name more explicit:

```suggestion
## Contributing to CodeQL CLI docs on `docs.github.com`
```

> **Note**: We have recently copied the CodeQL CLI documentation on [codeql.github.com](https://codeql.github.com/docs/codeql-cli/) to the public [github/docs](https://github.com/github/docs) repository so that this documentation will appear on the [GitHub Docs](https://docs.github.com/en/code-security/code-scanning) site. This includes all articles under "[Using the CodeQL CLI](https://codeql.github.com/docs/codeql-cli/using-the-codeql-cli/)" and "[CodeQL CLI reference](https://codeql.github.com/docs/codeql-cli/codeql-cli-reference/)" categories. 

Author: Yeah this looks a lot better!! Thanks :)

To contribute to these docs, which are located in the [`code-scanning`](https://github.com/github/docs/tree/main/content/code-security/code-scanning) directory, please refer to the [CONTRIBUTING.md](https://github.com/github/docs/blob/main/CONTRIBUTING.md) file in the `docs` repository.

> ## Contributing to CodeQL docs on `codeql.github.com`

Reviewer: Since this is the CodeQL repository, I think we probably need to lead with this section.

> To make changes to the remaining documentation on [codeql.github.com](https://codeql.github.com/docs/codeql-cli/), you can make changes to the documentation files using the GitHub UI, a codespace, or making changes locally, and then open a pull request for review. For more information about the format and structure of the CodeQL documentation on [codeql.github.com](https://codeql.github.com/docs/codeql-cli/), please see the [README](https://github.com/github/codeql/tree/main/docs/codeql#readme). 

Reviewer: I think that we need to change some of the links here. Suggest:

```suggestion
To make changes to the documentation on [codeql.github.com](https://codeql.github.com/docs/codeql-overview/), you can make changes to the documentation files using the GitHub UI, a codespace, or a local text editor, and then open a pull request for review. For more information about the format and structure of the CodeQL documentation on [codeql.github.com](https://codeql.github.com/docs/codeql-overview/), please see the [README](https://github.com/github/codeql/blob/main/docs/codeql/README.rst). 
```

> # Contributing to CodeQL docs

Reviewer 2: Minor: Typically, the file extension is lower case `docs/codeql/CONTRIBUTING.MD` -> `docs/codeql/CONTRIBUTING.md`

We welcome contributions to our CodeQL docs. Want to improve existing docs or add new information you think would be helpful? Then please go ahead and open a pull request!

## Contributing to CodeQL docs on `codeql.github.com`

To make changes to the documentation on [codeql.github.com](https://codeql.github.com/docs/codeql-overview/), you can make changes to the documentation files using the GitHub UI, a codespace, or a local text editor, and then open a pull request for review. For more information about the format and structure of the CodeQL documentation on [codeql.github.com](https://codeql.github.com/docs/codeql-overview/), please see the [README](https://github.com/github/codeql/blob/main/docs/codeql/README.rst).

## Contributing to CodeQL CLI docs on `docs.github.com`

> We are in the process of moving all documentation about the CodeQL CLI from [codeql.github.com](https://codeql.github.com/docs/codeql-cli/) to the public [github/docs](https://github.com/github/docs) repository so that this documentation is published on the [GitHub Docs](https://docs.github.com/en/code-security/code-scanning) site. This includes all articles under "[Using the CodeQL CLI](https://codeql.github.com/docs/codeql-cli/using-the-codeql-cli/)" and "[CodeQL CLI reference](https://codeql.github.com/docs/codeql-cli/codeql-cli-reference/)" categories. This will make it easier for code scanning users to find information about using CodeQL to query their codebases.

Reviewer: Just realized that I suggested mixed source/publication links in the first sentence here 🤦🏻‍♀️ 

Suggest we probably change this as follows:

```suggestion
We are in the process of moving all documentation about the CodeQL CLI from [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql) to the public [github/docs](https://github.com/github/docs) repository so that this documentation is published on the [GitHub Docs](https://docs.github.com/en/code-security/code-scanning) site. This includes all articles that are currently published under "[Using the CodeQL CLI](https://codeql.github.com/docs/codeql-cli/using-the-codeql-cli/)" and "[CodeQL CLI reference](https://codeql.github.com/docs/codeql-cli/codeql-cli-reference/)" categories on the CodeQL microsite. This will make it easier for code scanning users to find information about using CodeQL to query their codebases.
```

> **Note**: For a brief time, we will have source files for CodeQL CLI documentation in two locations. During this period we will not accept changes to the old files in the `codeql` repository, only to the new files in the `docs` repository. 

Reviewer 2: Are there any comments in the duplicated files on github/codeql mentioning that they shouldn't be touched?

Author: Yes, good point!  As part of [Batch 2](https://github.com/github/docs-content/issues/6930) we will be adding deprecation notes and directing users to the GitHub docs site. We could definitely add notes in these articles, but contributors will probably see the deprecation notes as well when they go to update them! We are hoping to follow up with Batch 2 very shortly so there is not too much time where both version of articles exist
